ygen/genstate.go

```diff
@@ -267,7 +292,7 @@ func (s *genState) findEnumSet(entries map[string]*yang.Entry, compressOCPaths b
 		case e.Type.Name == "union", len(e.Type.Type) > 0 && !builtin:
```

Comment:
Since you changed to != nil elsewhere, might as well do it here and the rest of the code. I think gofmt refactor should be able to do that for you.

Reply:
I think that this isn't robust though -- https://play.golang.org/p/ZqK-bUpJyM -- if anything initialised the slice as being empty, then it's not nil, and len(slice) > 0 is required to check that it wasn't initialised. In this case, e.Type.Type is initialised so non-nil. I refactored where it's internal to ygen.